File: languages/python/bloock/entity/identity_v2/decimal_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class DecimalAttribute(Attribute):
    @staticmethod
    def from_proto(a: proto.DecimalAttributeV2) -> DecimalAttribute:
        return DecimalAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.DecimalAttributeV2:
        return proto.DecimalAttributeV2(
            id=self.id,
            value=self.value,
        )

File: languages/js/src/entity/identity_v2/decimal_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { Attribute } from "./attribute";

/**
 * Represents an attribute with a decimal value.
 */
export class DecimalAttribute extends Attribute<number> {
  public toProto(): identityEntitiesProto.DecimalAttributeV2 {
    return identityEntitiesProto.DecimalAttributeV2.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.DecimalAttributeV2
  ): DecimalAttribute {
    return new DecimalAttribute(r.id, r.value);
  }
}
